src/main/java/de/teamlapen/vampirism/tileentity/AltarInspirationTileEntity.java

```diff
@@ -156,22 +150,23 @@ public void tick() {
         ritualTicksLeft--;
     }
 
-    private class InternalTank extends FluidTank {
+    @Override
+    public void onTankContentChanged() {
+        markDirty();
+    }
 
-        public InternalTank(int capacity) {
-            super(capacity);
-            setCanDrain(false);
-        }
+    private static class InternalTank extends FluidTankWithListener {
 
-        @Override
-        public boolean canFillFluidType(FluidStack fluid) {
-            return fluid != null && canFill() && ModFluids.blood.equals(fluid.getFluid());
+        private InternalTank(int capacity) {
+            super(capacity, fluidStack -> ModFluids.blood.isEquivalentTo(fluidStack.getFluid()));
+            setDrainable(false);
         }
 
-        public FluidStack doDrain(int maxDrain, boolean doDrain) {
-            this.setCanDrain(true);
-            FluidStack s = super.drain(maxDrain, doDrain);
-            this.setCanDrain(false);
+        @Override
+        public FluidStack drain(int maxDrain, FluidAction action) {
```

Comment:
If I understand this correctly, this makes the NotDrainableTank quite pointless :D If you override the drain method, then every call to drain this tank will make it drainable and therefore void the drainable check of the original method.

In the old code this was done in a new dedicated method doDrain which is only called by Vampirism